compiler/src/dotty/tools/dotc/typer/Nullables.scala

```diff
@@ -105,17 +105,50 @@ object Nullables with
    *  This is the case if the reference is a path to an immutable val, or if it refers
    *  to a local mutable variable where all assignments to the variable are _reachable_
    *  (in the sense of how it is defined in assignmentSpans).
+   *
+   *  A mutable vriable is trackable with following restrictions:
```

Comment:
I like having examples, but I think we should be a bit more explicit about what's happening here.

Maybe something like

```
When dealing with local mutable variables, there are two questions:
1. Whether to use flow typing at all on the local mutable variable. We use flow typing iff
    the variable is not assigned in a closure.

For example, in the following code `x` is assigned to by the closure `y`, so we do not do flow typing
on `x`.

(Paste first example here)

2. Whether to use flow typing on a specific _use_ of a local mutable variable. We only want to do flow typing
    on a use that belongs to the same method as the definition of the local variable. For example, in the following    code `x` is not assigned to by a closure, but we only use flow typing in one of the occurrences (because the other occurrence happens within a nested closure).

(Paste second example here)

```

Reply:
I will move the two example. It is a little difficult to add more than two examples, because each example is not short.